### ISSUE TITLE
inherit_mode is ignored for arrays other than Include/Exclude if also inheriting non-default configuration (with inherit_mode at cop level)

### ISSUE THREAD

**Issue by Drowze** [rubocop/rubocop#10026]
## Expected behavior

I expect to be able to merge my configuration with rubocop default configuration, even if I also inherit a non-default configuration.

## Actual behavior

At the project `.rubocop.yml`, if I define an array that already exists within rubocop's default configuration, my configuration overwrites' the default config, even if `inherit_mode` is set to merge on the cop level

## Steps to reproduce the problem

Given a hosted configuration ([gist here](https://gist.githubusercontent.com/Drowze/0ea51fa5bf3cbec9e280f2f913be57df/raw/eaa3ac4911b982ddf933e32e4f53d91c41cebb43/rubocop-test.yml)):
```yaml
AllCops:
  NewCops: enable

Naming/VariableNumber:
  EnforcedStyle: snake_case
  Exclude:
    - foo.rb
  Include:
    - bar.rb
```

And given `.rubocop.yml`:
```yaml
inherit_from:
  - https://gist.githubusercontent.com/Drowze/0ea51fa5bf3cbec9e280f2f913be57df/raw/eaa3ac4911b982ddf933e32e4f53d91c41cebb43/rubocop-test.yml

Naming/VariableNumber:
  inherit_mode:
    merge:
      - AllowedIdentifiers
      - Exclude
      - Include
  AllowedIdentifiers:
    - iso2
  Exclude:
    - test.rb
  Include:
    - another_test.rb
```

When I run `rubocop --debug --show-cops Naming/VariableNumber` I get this output:
(note that `AllowedIdentifiers` does not include `capture3` or any of the items defined at the [default configuration](https://raw.githubusercontent.com/rubocop/rubocop/master/config/default.yml))
```
For /Users/Drowze/r/rubocop_bug9: configuration from /Users/Drowze/r/rubocop_bug9/.rubocop.yml
configuration from /Users/Drowze/r/rubocop_bug9/.rubocop-https---gist-githubusercontent-com-Drowze-0ea51fa5bf3cbec9e280f2f913be57df-raw-eaa3ac4911b982ddf933e32e4f53d91c41cebb43-rubocop-test-yml
Default configuration from /Users/Drowze/.asdf/installs/ruby/2.7.3/lib/ruby/gems/2.7.0/gems/rubocop-1.19.0/config/default.yml
Naming/VariableNumber:
  Description: Use the configured style when numbering symbols, methods and variables.
  StyleGuide: "#snake-case-symbols-methods-vars-with-numbers"
  Enabled: true
  VersionAdded: '0.50'
  VersionChanged: '1.4'
  EnforcedStyle: snake_case
  SupportedStyles:
  - snake_case
  - normalcase
  - non_integer
  CheckMethodNames: true
  CheckSymbols: true
  AllowedIdentifiers:
  - iso2
  Exclude:
  - "/Users/Drowze/r/rubocop_bug9/foo.rb"
  - "/Users/Drowze/r/rubocop_bug9/test.rb"
  Include:
  - bar.rb
  - another_test.rb

Finished in 0.111557999975048 seconds
```

From my test I also found:
- if I move `inherit_mode` to the root level, AllowedIdentifiers is merged correctly
- if I do not inherit configuration (via `inherit_from` or `inherit_gem`), AllowedIdentifiers is merged correctly

## RuboCop version

```
$ rubocop -V
1.19.0 (using Parser 3.0.2.0, rubocop-ast 1.10.0, running on ruby 2.7.3 x86_64-darwin20)
```
